Dockerfile

```diff
@@ -0,0 +1,36 @@
+FROM node:16.13.1-alpine3.15@sha256:a2c7f8ebdec79619fba306cec38150db44a45b48380d09603d3602139c5a5f92 AS builder
+
+WORKDIR /app
+RUN [ "yarn", "set", "version", "1.22.17" ]
+COPY /src ./src
+COPY package.json ./
+COPY yarn.lock ./
+
+RUN yarn install
+RUN yarn build
+
+# 'yarn install --production' does not prune test packages which are necessary
+# to build the app. So delete nod_modules and reinstall only production packages.
+RUN [ "rm", "-rf", "node_modules" ]
+RUN yarn install --production
+
+FROM node:16.13.1-alpine3.15@sha256:a2c7f8ebdec79619fba306cec38150db44a45b48380d09603d3602139c5a5f92 as final
+
+RUN ["apk", "--no-cache", "upgrade"]
+RUN ["apk", "add", "--no-cache", "tini"]
+RUN [ "yarn", "set", "version", "1.22.17" ]
+
+WORKDIR /app
+# Copy in compile assets and deps from build container
+COPY --from=builder /app/node_modules ./node_modules
+COPY --from=builder /app/dist ./dist
+COPY --from=builder /app/src ./src
```

Comment:
Are we going to remove src/ before we take this up to production?

Reply:
Hey Andy, all for the removing what we don't need 👍  In this case though its a node app so the js that is loaded when the app starts is under the `src` directory (also third party modules from the `node_modules` directory), the start command which can be found in `package.json` runs `src/app.js`
```
GDS11321:di-ipv-cri-uk-passport-front dan.worth$ jq '.scripts.start' package.json
"node src/app.js"
```

The `app/dist` directory might be a bit of red herring if you were looking at that, it includes the compiled public assets like css etc.